pkg/apiserver/profiling/model.go

```diff
@@ -19,34 +23,80 @@ const (
 	TaskStateError TaskState = iota
 	TaskStateRunning
 	TaskStateFinish
-	TaskPartialFinish
+	TaskStatePartialFinish // Only valid for task group
+	TaskStateSkipped
 )
 
 type TaskRawDataType string
 
-const RawDataTypeProtobuf TaskRawDataType = "protobuf"
+const (
+	RawDataTypeProtobuf TaskRawDataType = "protobuf"
+	RawDataTypeText     TaskRawDataType = "text"
+)
+
+type (
+	TaskProfilingType     string
+	TaskProfilingTypeList []TaskProfilingType
+)
+
+func (r *TaskProfilingTypeList) Scan(src interface{}) error {
+	return json.Unmarshal([]byte(src.(string)), r)
+}
+
+func (r TaskProfilingTypeList) Value() (driver.Value, error) {
+	val, err := json.Marshal(r)
+	return string(val), err
+}
+
+const (
+	ProfilingTypeCPU       TaskProfilingType = "cpu"
+	ProfilingTypeHeap      TaskProfilingType = "heap"
+	ProfilingTypeGoroutine TaskProfilingType = "goroutine"
+	ProfilingTypeMutex     TaskProfilingType = "mutex"
+)
+
+var profilingTypeList = []TaskProfilingType{
+	ProfilingTypeCPU,
+	ProfilingTypeHeap,
+	ProfilingTypeGoroutine,
+	ProfilingTypeMutex,
+}
```

Comment:
How about using profiling type map like:
```
var profilingTypeMap = map[TaskProfilingType]struct{}{
	ProfilingTypeCPU: {},
	ProfilingTypeHeap: {},
	ProfilingTypeGoroutine: {},
	ProfilingTypeMutex: {},
}
```

So that we can easily check the valid profiling type
```
_, valid := profilingTypeMap[profilingType]
...
```